pkg/internal/controller/controller.go

```diff
@@ -267,6 +267,15 @@ func (c *Controller) processNextWorkItem(ctx context.Context) bool {
 	return true
 }
 
+// Len returns the current queue length, for informational purposes only. You
+// shouldn't e.g. gate a call to Add() or Get() on Len() being a particular
+// value, that can't be synchronized properly.
+func (c *Controller) Len() int {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	return c.Queue.Len()
```

Comment:
You have to check whether `c.Queue` is nil, for the field is assigned in later `Start()`.

Also I'm not sure if we really need to export the length of queue. I will leave my comment in the issue.